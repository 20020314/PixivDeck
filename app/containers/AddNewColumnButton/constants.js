// @flow
import type { ADD_USER_ILLUST_TYPE } from './actionTypes.js'

export const ADD_USER_ILLUST: ADD_USER_ILLUST_TYPE =
  'AddNewColumnButton/ADD_USER_ILLUST'
