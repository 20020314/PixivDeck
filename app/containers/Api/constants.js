// @flow
import type { API_REQUEST_SUCCESS_TYPE } from './actionTypes.js'

export const API_REQUEST_SUCCESS: API_REQUEST_SUCCESS_TYPE =
  'Api/API_REQUEST_SUCCESS'
