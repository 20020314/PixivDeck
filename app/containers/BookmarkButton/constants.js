// @flow
import type {
  ADD_BOOKMARK_REQUEST_TYPE,
  ADD_BOOKMARK_SUCCESS_TYPE,
  ADD_BOOKMARK_FAILER_TYPE,
} from './actionTypes.js'

export const ADD_BOOKMARK_REQUEST: ADD_BOOKMARK_REQUEST_TYPE =
  'BookmarkButton/ADD_BOOKMARK_REQUEST'
export const ADD_BOOKMARK_SUCCESS: ADD_BOOKMARK_SUCCESS_TYPE =
  'BookmarkButton/ADD_BOOKMARK_SUCCESS'
export const ADD_BOOKMARK_FAILER: ADD_BOOKMARK_FAILER_TYPE =
  'BookmarkButton/ADD_BOOKMARK_FAILER'
