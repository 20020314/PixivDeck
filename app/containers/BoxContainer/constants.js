// @flow
import type { OPEN_PIXIV_TYPE } from './actionTypes.js'

export const OPEN_PIXIV: OPEN_PIXIV_TYPE = 'BoxContainer/OPEN_PIXIV'
