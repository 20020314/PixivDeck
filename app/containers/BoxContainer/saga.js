// @flow
import { shell } from 'electron'
import { Actions } from './actionTypes'
import { call, takeEvery, type IOEffect } from 'redux-saga/effects'

function* openPixiv({ id }: { id: number }): Generator<IOEffect, void, *> {
  yield call(
    shell.openExternal,
    `http://www.pixiv.net/member_illust.php?mode=medium&illust_id=${id}`
  )
}

export default function* root(): Generator<IOEffect, void, *> {
  yield takeEvery(Actions.OPEN_PIXIV, openPixiv)
}
