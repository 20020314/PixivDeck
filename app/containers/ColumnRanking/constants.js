// @flow
import type {
  ADD_RANKING_COLUMN_TYPE,
  ADD_RANKING_COLUMN_SUCCESS_TYPE,
  SET_NEXT_URL_TYPE,
  FETCH_RANKING_TYPE,
  FETCH_RANKING_SUCCESS_TYPE,
  FETCH_RANKING_FAILRE_TYPE,
} from './actionTypes.js'

export const ADD_RANKING_COLUMN: ADD_RANKING_COLUMN_TYPE =
  'ColumnRanking/ADD_COLUMN'
export const ADD_RANKING_COLUMN_SUCCESS: ADD_RANKING_COLUMN_SUCCESS_TYPE =
  'ColumnRanking/ADD_COLUMN_SUCCESS'
export const SET_NEXT_URL: SET_NEXT_URL_TYPE = 'ColumnRanking/SET_NEXT_URL'
export const FETCH_RANKING: FETCH_RANKING_TYPE = 'ColumnRanking/FETCH_RANKING'
export const FETCH_RANKING_SUCCESS: FETCH_RANKING_SUCCESS_TYPE =
  'ColumnRanking/FETCH_RANKING_SUCCESS'
export const FETCH_RANKING_FAILRE: FETCH_RANKING_FAILRE_TYPE =
  'ColumnRanking/FETCH_RANKING_FAILRE'
