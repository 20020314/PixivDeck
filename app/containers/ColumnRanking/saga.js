// @flow
import union from 'lodash/union'
import { addColumn } from 'containers/ColumnManager/actions'
import { getRequest } from '../../api/client'
import * as Actions from './constants'
import * as actions from './actions'
import type { Mode } from './reducer'
import { makeSelectColumn, makeSelectModes } from './selectors'
import { put, select, call, takeEvery } from 'redux-saga/effects'

function* addRakingColumn({ mode }: { mode: Mode }) {
  const modes: Array<?Mode> = yield select(makeSelectModes())
  if (modes.every(v => v !== mode)) {
    yield put(actions.addRankingColumnSuccess(mode))
  }

  yield put(addColumn(`ranking-${mode}`, { columnId: mode, type: 'RANKING' }))
}

type Props = { id: Mode }

function* fetchRanking(props: Props) {
  const { id } = props
  const { illustIds } = yield select(makeSelectColumn(), props)

  try {
    const response = yield call(getRequest, `/v1/illust/ranking?mode=${id}`)
    const { result } = response

    yield put(actions.setNextUrl(id, result.nextUrl))

    const nextIds = union(illustIds, result.illusts)
    yield put(actions.fetchRankingSuccess(id, response, nextIds))
  } catch (err) {
    yield put(actions.fetchRankingFailre(id))
  }
}

function* fetchNextRanking(props: Props) {
  const { id } = props
  const { illustIds, nextUrl } = yield select(makeSelectColumn(), props)

  try {
    if (!nextUrl) {
      return
    }

    const response = yield call(getRequest, nextUrl)
    const { result } = response

    yield put(actions.setNextUrl(id, result.nextUrl))

    const nextIds = union(illustIds, result.illusts)
    yield put(actions.fetchRankingSuccess(id, response, nextIds))
  } catch (err) {
    yield put(actions.fetchNextRankingFailre(id))
  }
}

export default function* root(): Generator<*, void, void> {
  yield takeEvery(Actions.ADD_RANKING_COLUMN, addRakingColumn)
  yield takeEvery(Actions.FETCH_RANKING, fetchRanking)
  yield takeEvery(Actions.FETCH_NEXT_RANKING, fetchNextRanking)
}
