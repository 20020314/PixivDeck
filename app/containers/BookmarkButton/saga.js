// @flow
import { call, put, takeEvery, type IOEffect } from 'redux-saga/effects'
import { get } from '../Api/sagas'
import * as api from 'containers/Api/sagas'
import * as Actions from './constants'
import * as actions from './actions'
import type { Restrict } from './types'

type Props = {
  id: number,
  restrict: Restrict,
}

export function* bookmark({ id, restrict }: Props): Generator<*, void, void> {
  try {
    yield call(api.post, '/v2/illust/bookmark/add', { illustId: id, restrict })

    yield put(actions.addBookmarkSuccess(id, restrict))
  } catch (err) {
    yield put(actions.addBookmarkFailer(id, err))
  }
}

export function* deleteTask({ id }: Props): Generator<*, void, void> {
  try {
    yield call(api.post, '/v1/illust/bookmark/delete', { illustId: id })
    yield put(actions.deleteBookmarkSuccess(id))
  } catch (err) {
    yield put(actions.deleteBookmarkFailer(id, err))
  }
}

function* success({ id }: Props) {
  try {
    yield call(get, `/v1/illust/detail?illust_id=${id}`, true)
  } catch (err) {}
}

function* root(): Generator<IOEffect, void, *> {
  yield takeEvery(Actions.ADD_BOOKMARK_REQUEST, bookmark)
  yield takeEvery(Actions.DELETE_BOOKMARK_REQUEST, deleteTask)
  yield takeEvery(
    [Actions.ADD_BOOKMARK_SUCCESS, Actions.DELETE_BOOKMARK_SUCCESS],
    success
  )
}

export default root
