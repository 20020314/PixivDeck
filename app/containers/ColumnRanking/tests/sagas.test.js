// @flow
import { takeEvery, take, call } from 'redux-saga/effects'
import * as sagas from '../saga'
import * as constants from '../constants'

test('root', () => {
  const gen = sagas.default()
  expect(gen.next().value).toEqual(
    takeEvery(constants.ADD_COLUMN, sagas.addColumn)
  )
  expect(gen.next().value).toEqual(takeEvery(constants.FETCH, sagas.fetch))
  expect(gen.next().value).toMatchSnapshot()
})

test('watchNewIllust', () => {
  const gen = sagas.watchNewIllust()
  expect(gen.next().value).toEqual(take(constants.START_WATCH))
  expect(gen.next({ id: 1 }).value).toMatchSnapshot()
  expect(gen.next().value).toMatchSnapshot()
  // $FlowFixMe
  expect(gen.next().value).toEqual(call(sagas.fetchNew, { id: 1 }))
})
