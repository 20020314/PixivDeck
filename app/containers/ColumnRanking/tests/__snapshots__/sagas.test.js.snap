// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`watchNewIllust 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [
      Object {
        "id": 1,
      },
    ],
    "selector": [Function],
  },
}
`;

exports[`watchNewIllust 2`] = `
Promise {
  "@@redux-saga/CANCEL_PROMISE": [Function],
}
`;
